File: Python/Baekjoon/[Gold ll] 공항 (10775번).py
#!/usr/bin/env python
# coding: utf-8

# In[ ]:


import sys
input = sys.stdin.readline

def find(x):
    if parent[x] == x:
        return x
    parent[x]=find(parent[x])
    return parent[x]

def union(a, b): 
    a = find(a) 
    b = find(b) 

import sys
input = sys.stdin.readline

def find(x):
    if parent[x] == x:
        return x
    parent[x]=find(parent[x])
    return parent[x]

G = int(input())
P = int(input())

parent = [i for i in range(G+1)]
count = 0
for i in range(P):
    plane = int(input())
    if find(plane) == 0:
        break
    parent[find(plane)] = parent[find(plane)-1]
    count+=1
    
print(count)

File: Python/Baekjoon/[Gold ll] 공항 (10775번).java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class Main{
    static int[] parent;
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int p = Integer.parseInt(br.readLine());
        int count = 0;
        parent = new int[n+1];
        for (int i = 0; i<n+1; i++){
            parent[i] = i;
        }
        for(int i = 0; i<p; i++){
            int plane = Integer.parseInt(br.readLine());
            if(parent[find(plane)] == 0){
                break;
            }
            parent[find(plane)] = find(plane)-1;
            count++;
        }
        System.out.println(count);
    }
    public static int find(int x){
        if (parent[x] == x){
            return x;
        }
        parent[x] = find(parent[x]);
        return parent[x];
    }
}